### ISSUE TITLE
After TLSActor failure, system.terminate() hangs and times out

### ISSUE THREAD

**Issue by raboof** [akka/akka#29162]
Reproducer at https://github.com/akka/akka/compare/master...raboof:system-terminate-times-out-after-TLSActor-failure

**Comment by ignasi35** [comment]
More info at https://github.com/akka/akka-http/pull/2983#issuecomment-591912391

**Comment by raboof** [comment]
> More info at akka/akka-http#2983 (comment)

I don't think that is related: there it was not the 'terminate' that timed out, but the `MaterializerState.streamSnapshots`, which was filed as #28691 and fixed in #29047

**Comment by jrudolph** [comment]
Is there still something left to do here?

**Comment by raboof** [comment]
I don't think we've really diagnosed why system-terminate-times-out-after-TLSActor-failure hangs yet - perhaps rebase it on master and see if it still fails?

**Comment by jrudolph** [comment]
> I don't think we've really diagnosed why system-terminate-times-out-after-TLSActor-failure hangs yet - perhaps rebase it on master and see if it still fails?

Ah, was confused about the above comments which was confused itself...

**Comment by raboof** [comment]
It *might* be fixed by your recent TLS1.3 fixes though, so might be worth checking whether the reproducer still shows the issue after rebasing
